ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationSidebar.tsx

```diff
@@ -106,83 +90,91 @@ type ParentDescription = {
 
 type NavDescription = LinkDescription | ParentDescription;
 
-const navDescriptions: NavDescription[] = [
-    {
-        type: 'link',
-        content: 'Dashboard',
-        path: dashboardPath,
-        routeKey: 'dashboard',
-    },
-    {
-        type: 'parent',
-        title: 'Network',
-        key: keyForNetwork,
-        children: [
-            {
-                type: 'link',
-                content: 'Network Graph',
-                path: networkBasePath,
-                routeKey: 'network-graph',
-            },
-            {
-                type: 'link',
-                content: 'Listening Endpoints',
-                path: listeningEndpointsBasePath,
-                routeKey: 'listening-endpoints',
-            },
-        ],
-    },
-    {
-        type: 'link',
-        content: 'Violations',
-        path: violationsBasePath,
-        routeKey: 'violations',
-    },
-    // Compliance with divided navigation.
-    // /*
-    {
-        type: 'parent',
-        title: (navDescriptionsFiltered) =>
-            navDescriptionsFiltered.some(
-                (navDescription) =>
-                    navDescription.type === 'link' && navDescription.routeKey === 'compliance'
-            )
-                ? 'Compliance (2.0)'
-                : 'Compliance',
-        key: keyForCompliance2,
-        children: [
-            {
-                type: 'link',
-                content: 'Coverage',
-                path: complianceEnhancedCoveragePath,
-                routeKey: 'compliance-enhanced',
-            },
-            {
-                type: 'link',
-                content: 'Schedules',
-                path: complianceEnhancedSchedulesPath,
-                routeKey: 'compliance-enhanced',
-            },
-        ],
-    },
-    {
-        type: 'link',
-        content: (navDescriptionsFiltered) =>
-            navDescriptionsFiltered.some(
-                (navDescription) =>
-                    navDescription.type === 'parent' && navDescription.key === keyForCompliance2
-            )
-                ? 'Compliance (1.0)'
-                : 'Compliance',
-        path: complianceBasePath,
-        routeKey: 'compliance',
-        isActive: (pathname) =>
-            Boolean(matchPath(pathname, complianceBasePath)) &&
-            !matchPath(pathname, [complianceEnhancedCoveragePath, complianceEnhancedSchedulesPath]),
-    },
-    // */
-    // Compliance with unified navigation.
-    /*
+function getNavDescriptions(isFeatureFlagEnabled: IsFeatureFlagEnabled): NavDescription[] {
+    const vulnMgmt2GaContentPredicate =
+        (techPreviewContent: string | ReactElement, gaContent: string | ReactElement) => () =>
+            isFeatureFlagEnabled('ROX_VULN_MGMT_2_GA') ? gaContent : techPreviewContent;
+
+    return [
```

Comment:
The logic for displaying the "GA" variations above was fairly complication, and with this change, no longer works. For simplicity and flexibility, we can now pass the `isFeatureFlagEnabled` function to a getter function for use in the closure instead of relying on the presence of other navigation items, when applicable.

Reply:
I pushed another commit to make this even less complicated (thanks @pedrottimark !)